src/backends/linux_raw.rs

```diff
@@ -0,0 +1,127 @@
+//! Implementation for Linux / Android using `asm!`-based syscalls.
+use crate::{Error, MaybeUninit};
+
+pub use crate::util::{inner_u32, inner_u64};
+
+#[cfg(not(any(target_os = "android", target_os = "linux")))]
```

Comment:
I'm trying to figure out what the constraints are around directly invoking syscalls on Android. I can try to check with some people at work, but it's unclear if bionic libc (the libc shipped with Android) does any special handling for syscalls. If the only demand for this is on `linux` targets, it might just be best to initially exclude Android.

Reply:
Yeah, it would be nice to verify that raw syscalls work on Android without any surprises.

Here is a relevant `rustix` discussion: https://github.com/bytecodealliance/rustix/issues/1095 No one has mentioned any fundamental technical obstacles to using raw syscalls and it looks like some people [use them in practice](https://github.com/bytecodealliance/rustix/pull/1090#issuecomment-2248330606).

Reply:
Since users have to explicitly enable the backend, I think we can leave it as-is and potentially disable it later if it will be found problematic on Android for some reason.